### ISSUE TITLE
Openstack-novaLXD: no Deploy options are visible

### ISSUE THREAD

**Issue by abatti** [conjure-up/conjure-up#1420]
For Openstack-novaLXD i can see only Config options for all item.
There are no Deploy options

## Report

Thank you for trying **conjure-up**! Before reporting a bug please make sure you've gone through this checklist:

- [ ] Is this problem already documented at https://docs.ubuntu.com/conjure-up/en/troubleshoot#common-spell-problems
- [ ] Is conjure-up running inside a virtual machine?
- [ ] Is this problem reproducible with `sudo snap refresh conjure-up --edge`?

## Please provide the output of the following commands

```
which juju
/snap/bin/juju

juju version
2.3.7-xenial-amd64

which conjure-up
/snap/bin/conjure-up

conjure-up --version
conjure-up 2.5.6

which lxc
/snap/bin/lxc

/snap/bin/lxc config show
/snap/bin/lxc version

cat /etc/lsb-release
```

Please attach tarball of **~/.cache/conjure-up**:

```
tar cvzf conjure-up.tar.gz ~/.cache/conjure-up
```

## Sosreport

Please attach a sosreport:

```
sudo apt install sosreport
sosreport
```

The resulting output file can be attached to this issue.

### MAAS Users
Which version of MAAS?

## Commands ran

Please outline what commands were run to install and execute conjure-up:

**Comment by battlemidget** [comment]
What type of deploy options are you looking for?

**Comment by abatti** [comment]
0)
The "deploy options"  (one for each item) that you can see in the first
image at
https://insights.ubuntu.com/2018/02/02/tutorial-install-single-server-openstack-with-conjure-up

In "my" conjure-up (version 3.0.0 via snap) i can see only "Configure
options"  (one for each item)
and single "Deploy" button (only the single term "Deploy", not "deploy all
remainings...") at bottom.


In the meantime I tried other actions:
1)
I pushed deploy but I got the "typical error": Unable to bootstrap (cloud
type:localhost)
but it was my fault: there was an error in my proxy configuration

2)
after restarted conjure-up I pushed deploy again, but this time juju seems
to bootstrap for the eternity:
I see only the animated widget but NO logs or other messages.


So I ask you:
About 0): maybe conjure-up gui has changed and "deploy options" (one for
each item) are no more available?

About 2): maybe I'm doing the right thing and I have only to wait for juju
to terminate his process?
How can I verify that juju is working?

Thank you all.
AB







2018-04-27 21:46 GMT+02:00 Adam Stokes <notifications@github.com>:

**Comment by battlemidget** [comment]
> So I ask you:
> About 0): maybe conjure-up gui has changed and "deploy options" (one for
> each item) are no more available?

Yea we've updated the gui a bit, we use native juju deploy now so no need for the placement options as the bundle now gets rebuilt to a format juju understands. Also we don't deploy applications individually any longer so we only need a single deploy button now.

> 
> About 2): maybe I'm doing the right thing and I have only to wait for juju
> to terminate his process?
> How can I verify that juju is working?

Make sure you've read through https://docs.conjure-up.io/devel/en/#users-of-lxd and verify you have lxd setup. Once that's done you should be able to deploy.

Thanks

**Comment by abatti** [comment]
Thank you for your confirm about single Deploy Button and no (more) deploy options.
BUT...
For 2) I've read the guide and tested lxc with launch,exec,stop,delete and they worked correctly.

When I try to use conjure-up, last log lines are:

[DEBUG] conjure-up/openstack-novalxd - telemetry.py:31 - Juju Bootstrap: Started 
[DEBUG] conjure-up/openstack-novalxd - juju.py:213 - Bootstrapping to set region: {}
[DEBUG] conjure-up/openstack-novalxd - juju.py:249 - bootstrap cmd: ['/snap/bin/juju', 'bootstrap', 'localhost/localhost', 'conjure-up-localhost-42b', '--default-model', 'conjure-openstack-novalx-f36', '--config', 'image-stream=daily', '--bootstrap-series', 'xenial']
[DEBUG] conjure-up/openstack-novalxd - events.py:53 - Awaiting Bootstrapped at conjureup/controllers/bootstrap/gui.py:43

I have waited for an hour but juju seems stucked (even if I can see the animated widget on the screen)
and I don't understand where...
Can I try other actions??

**Comment by abatti** [comment]
I've tried to run juju without conjure-up  BUT juju stuck in "Running machine configuration script..."

This is the log in debug mode:

ab@ab-Precision-Tower-3620:~$ juju bootstrap --debug
13:20:40 INFO  juju.cmd supercommand.go:56 running juju [2.3.3 gc go1.8]
13:20:40 DEBUG juju.cmd supercommand.go:57   args: []string{"/snap/conjure-up/987/bin/juju", "bootstrap", "--debug"}
Clouds
aws
aws-china
aws-gov
azure
azure-china
cloudsigma
google
joyent
localhost
oracle
rackspace

Select a cloud [localhost]: 

Enter a name for the Controller [localhost-localhost]: aMano

13:20:54 DEBUG juju.tools.lxdclient client.go:229 Using environment LXD_DIR as socket path: "/var/snap/lxd/common/lxd"
13:20:54 DEBUG juju.tools.lxdclient client.go:252 connecting to LXD remote "local": "unix:///var/snap/lxd/common/lxd/unix.socket"
13:20:54 INFO  juju.tools.lxdclient client.go:306 using LXD API version "1.0"
13:20:54 INFO  juju.tools.lxdclient client.go:355 LXD "default" profile uses network bridge "lxdbr0"
13:20:54 INFO  cmd provider.go:192 Resolved LXD host address on bridge lxdbr0: 10.215.46.1:8443
13:20:54 INFO  cmd cloudcredential.go:51 no credentials found, checking environment
13:20:54 DEBUG juju.tools.lxdclient client.go:229 Using environment LXD_DIR as socket path: "/var/snap/lxd/common/lxd"
13:20:54 DEBUG juju.tools.lxdclient client.go:252 connecting to LXD remote "local": "unix:///var/snap/lxd/common/lxd/unix.socket"
13:20:54 INFO  juju.tools.lxdclient client.go:306 using LXD API version "1.0"
13:20:54 INFO  juju.tools.lxdclient client.go:355 LXD "default" profile uses network bridge "lxdbr0"
13:20:54 DEBUG juju.cmd.juju.commands bootstrap.go:838 authenticating with region "localhost" and credential "localhost" ()
13:20:54 DEBUG juju.cmd.juju.commands bootstrap.go:963 provider attrs: map[]
13:20:54 INFO  cmd authkeys.go:114 Adding contents of "/home/ab/.local/share/juju/ssh/juju_id_rsa.pub" to authorized-keys
13:20:54 DEBUG juju.cmd.juju.commands bootstrap.go:1019 preparing controller with config: map[disable-network-management:false logging-config: automatically-retry-hooks:true update-status-hook-interval:5m test-mode:false apt-no-proxy: apt-https-proxy: apt-ftp-proxy: logforward-enabled:false proxy-ssh:false uuid:41f9f6e0-6ed4-408b-8b58-824636b89212 agent-metadata-url: name:controller firewall-mode:instance fan-config: egress-subnets: type:lxd enable-os-upgrade:true ignore-machine-addresses:false ftp-proxy: development:false image-stream:released net-bond-reconfigure-delay:17 max-action-results-size:5G http-proxy: transmit-vendor-metrics:true agent-stream:released apt-http-proxy: resource-tags: https-proxy: container-inherit-properties: apt-mirror: enable-os-refresh-update:true cloudinit-userdata: provisioner-harvest-mode:destroyed ssl-hostname-verification:true no-proxy:127.0.0.1,localhost,::1 max-action-results-age:336h container-networking-method: max-status-history-size:5G authorized-keys:ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC9GfS+ePbvQIDYkHILCm1cw7171FDUJKax5MaWEBlI+eNzbaROwSH2APh+M6yEslaW9LAq3BAQe1CK/VCgeV2kpNEhysvJvTUOlz4ftmBfZR+iUWwcob5zAIToRBFswAlYoY12044SuXJgZsbKAHpccIDQpaRTbHOdIEK4Cgq3bvIhdWqsByICJr8esJxujDnXxzcNpnGtmFttcHorl6FiTXLmbsP1qdbVqOEe+NpeS2j7TqEJu4o70nPC9ZrQAr/NtcAyKLbV0dN45qsVu7z68Nq9JOaGZXi/0IXXzFo4MaIng9qBoOxfO+3DVVyz3HzjhjzFhalhv17J+sFS1nZZ juju-client-key
 max-status-history-age:336h image-metadata-url: default-series:bionic]
13:20:54 DEBUG juju.tools.lxdclient client.go:229 Using environment LXD_DIR as socket path: "/var/snap/lxd/common/lxd"
13:20:54 DEBUG juju.tools.lxdclient client.go:252 connecting to LXD remote "remote": "10.215.46.1:8443"
13:20:54 INFO  juju.tools.lxdclient client.go:306 using LXD API version "1.0"
13:20:55 INFO  cmd bootstrap.go:511 Creating Juju controller "aMano" on localhost/localhost
13:20:55 INFO  juju.cmd.juju.commands bootstrap.go:572 combined bootstrap constraints: 
13:20:55 DEBUG juju.environs.bootstrap bootstrap.go:199 model "controller" supports service/machine networks: false
13:20:55 DEBUG juju.environs.bootstrap bootstrap.go:201 network management by juju enabled: true
13:20:55 INFO  cmd bootstrap.go:233 Loading image metadata
13:20:55 INFO  cmd bootstrap.go:303 Looking for packaged Juju agent version 2.3.3 for amd64
13:20:55 INFO  juju.environs.bootstrap tools.go:72 looking for bootstrap agent binaries: version=2.3.3
13:20:55 DEBUG juju.environs.tools tools.go:102 finding agent binaries in stream: "released"
13:20:55 DEBUG juju.environs.tools tools.go:104 reading agent binaries with major.minor version 2.3
13:20:55 DEBUG juju.environs.tools tools.go:112 filtering agent binaries by version: 2.3.3
13:20:55 DEBUG juju.environs.tools tools.go:118 filtering agent binaries by architecture: amd64
13:20:55 DEBUG juju.environs.tools urls.go:109 trying datasource "keystone catalog"
13:20:55 DEBUG juju.environs.simplestreams simplestreams.go:683 using default candidate for content id "com.ubuntu.juju:released:tools" are {20161007 mirrors:1.0 content-download streams/v1/cpc-mirrors.sjson []}
13:20:56 INFO  juju.environs.bootstrap tools.go:74 found 15 packaged agent binaries
13:20:56 INFO  cmd bootstrap.go:389 Starting new instance for initial controller
13:20:56 INFO  cmd environ.go:152 To configure your system to better support LXD containers, please see: https://github.com/lxc/lxd/blob/master/doc/production-setup.md
13:20:56 INFO  cmd bootstrap.go:151 Launching controller instance(s) on localhost/localhost...
13:20:56 DEBUG juju.provider.lxd environ_broker.go:35 StartInstance: "0", bionic
13:20:56 DEBUG juju.cloudconfig.instancecfg instancecfg.go:835 Setting numa ctl preference to false
13:20:56 DEBUG juju.environs imagemetadata.go:112 obtained image datasource "default cloud images"
13:20:56 DEBUG juju.environs imagemetadata.go:112 obtained image datasource "default ubuntu cloud images"
13:20:56 DEBUG juju.provider.lxd environ_broker.go:107 LXD requires https://, using: https://cloud-images.ubuntu.com/releases/
13:20:56 DEBUG juju.service discovery.go:63 discovered init system "systemd" from series "bionic"
13:20:56 DEBUG juju.provider.lxd environ_broker.go:225 LXD user data; 3572 bytes
13:20:56 INFO  juju.provider.lxd environ_broker.go:206 starting instance "juju-b89212-0" (image "juju/bionic/amd64")...
13:21:03 INFO  juju.provider.lxd environ_broker.go:51 started instance "juju-b89212-0"
13:21:03 INFO  cmd bootstrap.go:225  - juju-b89212-0 (arch=amd64)          
13:21:03 INFO  juju.environs.bootstrap bootstrap.go:638 newest version: 2.3.3
13:21:03 INFO  juju.environs.bootstrap bootstrap.go:653 picked bootstrap agent binary version: 2.3.3
13:21:03 INFO  cmd bootstrap.go:425 Installing Juju agent on bootstrap instance
13:21:03 INFO  cmd bootstrap.go:517 Fetching Juju GUI 2.12.1
13:21:03 DEBUG juju.cloudconfig.instancecfg instancecfg.go:835 Setting numa ctl preference to false
Waiting for address
Attempting to connect to 10.215.46.146:22
13:21:14 INFO  cmd bootstrap.go:333 Connected to 10.215.46.146
13:21:14 INFO  juju.cloudconfig userdatacfg_unix.go:460 Fetching agent: curl -sSfw 'agent binaries from %{url_effective} downloaded: HTTP %{http_code}; time %{time_total}s; size %{size_download} bytes; speed %{speed_download} bytes/s ' --retry 10 -o $bin/tools.tar.gz <[https://streams.canonical.com/juju/tools/agent/2.3.3/juju-2.3.3-ubuntu-amd64.tgz]>
13:21:14 INFO  cmd bootstrap.go:403 Running machine configuration script...

AFTER 2 hours of waiting I gave up

Any ideas???

**Comment by abatti** [comment]
Finally... Openstack installed....
The solution was to switch to a "free IP" (in my organization) that does not use HTTP PROXY.
BUT I still don't understand why.
I'm sure to have configured HTTP PROXY correctly.
I've used http_proxy, https_proxy and no_proxy IN MY /etc/environment
and I've used same configuration in "lxc config" (lxc config set core.proxy_http ... and the others)
It would be nice if:
1) someone could tell me which other configurations are necessary to use/install conjure-up/openstack
working with a HTTP PROXY
2) someone could improve the level of logging, adding details about possible problems using a HTTP PROXY

Thank you all
AB
